src/Plugins/ComplexCore/test/CreateDataArrayTest.cpp

```diff
@@ -18,9 +18,69 @@ TEST_CASE("Instantiate CreateDataArray Filter", "[ComplexCore][CreateDataArray][
+  args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
   args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
 
   auto result = filter.execute(ds, args);
   REQUIRE(result.result.valid());
 }
+
+TEST_CASE("CreateDataArray(Invalid Parameters)", "[ComplexCore][CreateDataArray]")
+{
+  static constexpr uint64 k_NComp = 3;
+  static constexpr uint64 k_NumTuples = 25;
+
+  DataPath dataPath({"foo"});
+  {
+    CreateDataArray filter;
+    DataStructure ds;
+    Arguments args;
+    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::uint16));
+    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(k_NComp));
+    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
+    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
+    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("-1"));
+
+    auto result = filter.execute(ds, args);
+    REQUIRE(result.result.invalid());
+  }
+  {
+    CreateDataArray filter;
+    DataStructure ds;
+    Arguments args;
+    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::int8));
+    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(k_NComp));
+    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
+    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
+    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1024"));
+
+    auto result = filter.execute(ds, args);
+    REQUIRE(result.result.invalid());
+  }
+  {
+    CreateDataArray filter;
+    DataStructure ds;
+    Arguments args;
+    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::float32));
+    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(0));
+    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
+    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
+    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1"));
+
+    auto result = filter.execute(ds, args);
+    REQUIRE(result.result.invalid());
+  }
+  {
+    CreateDataArray filter;
+    DataStructure ds;
+    Arguments args;
+    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::float32));
+    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(1));
+    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(0));
+    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
+    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1"));
+
+    auto result = filter.execute(ds, args);
+    REQUIRE(result.result.invalid());
+  }
+}
```

Comment:
```suggestion
TEST_CASE("CreateDataArray(Invalid Parameters)", "[ComplexCore][CreateDataArray]")
{
  static constexpr uint64 k_NComp = 3;
  static constexpr uint64 k_NumTuples = 25;

  DataPath dataPath({"foo"});

  CreateDataArray filter;
  DataStructure ds;
  Arguments args;

  SECTION("Section1")
  {
    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::uint16));
    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(k_NComp));
    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("-1"));

    auto result = filter.execute(ds, args);
    REQUIRE(result.result.invalid());
  }
  SECTION("Section2")
  {
    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::int8));
    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(k_NComp));
    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1024"));

    auto result = filter.execute(ds, args);
    REQUIRE(result.result.invalid());
  }
  SECTION("Section3")
  {
    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::float32));
    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(0));
    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(k_NumTuples));
    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1"));

    auto result = filter.execute(ds, args);
    REQUIRE(result.result.invalid());
  }
  SECTION("Section4")
  {
    args.insert(CreateDataArray::k_NumericType_Key, std::make_any<NumericType>(NumericType::float32));
    args.insert(CreateDataArray::k_NumComps_Key, std::make_any<uint64>(1));
    args.insert(CreateDataArray::k_NumTuples_Key, std::make_any<uint64>(0));
    args.insert(CreateDataArray::k_DataPath_Key, std::make_any<DataPath>(dataPath));
    args.insert(CreateDataArray::k_InitilizationValue_Key, std::make_any<std::string>("1"));

    auto result = filter.execute(ds, args);
    REQUIRE(result.result.invalid());
  }
}
```
The test blocks here are the typical use case for [Catch2 sections](https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md).